src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/security/AuthorisedCaseSearchOperation.java

```diff
@@ -53,36 +57,56 @@ public AuthorisedCaseSearchOperation(
     }
 
     @Override
-    public CaseSearchResult execute(CaseSearchRequest caseSearchRequest) {
-        return authorisedCaseDefinitionDataService
-            .getAuthorisedCaseType(caseSearchRequest.getCaseTypeId(), CAN_READ)
-            .map(caseType -> searchCasesAndFilterFieldsByAccess(caseType, caseSearchRequest))
-            .orElse(CaseSearchResult.EMPTY);
+    public CaseSearchResult execute(CrossCaseTypeSearchRequest searchRequest) {
+        List<CaseType> authorisedCaseTypes = getAuthorisedCaseTypes(searchRequest);
+        CrossCaseTypeSearchRequest authorisedSearchRequest = createAuthorisedSearchRequest(authorisedCaseTypes, searchRequest);
+
+        return searchCasesAndFilterFieldsByAccess(authorisedCaseTypes, authorisedSearchRequest);
     }
 
-    private CaseSearchResult searchCasesAndFilterFieldsByAccess(CaseType caseType, CaseSearchRequest caseSearchRequest) {
-        CaseSearchResult result = caseSearchOperation.execute(caseSearchRequest);
-        filterFieldsByAccess(caseType, result.getCases());
-        return result;
+    private List<CaseType> getAuthorisedCaseTypes(CrossCaseTypeSearchRequest searchRequest) {
+        return searchRequest.getCaseTypeIds()
+            .stream()
+            .map(caseTypeId -> authorisedCaseDefinitionDataService.getAuthorisedCaseType(caseTypeId, CAN_READ).orElse(null))
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
     }
 
-    private void filterFieldsByAccess(CaseType caseType, List<CaseDetails> cases) {
-        cases.forEach(caseDetails -> {
-            filterCaseFieldsByAclAccess(caseType, caseDetails);
-            filterCaseFieldsBySecurityClassification(caseDetails);
-        });
+    private CrossCaseTypeSearchRequest createAuthorisedSearchRequest(List<CaseType> authorisedCaseTypes, CrossCaseTypeSearchRequest originalSearchRequest) {
+        List<String> authorisedCaseTypeIds = authorisedCaseTypes.stream().map(CaseType::getId).collect(Collectors.toList());
+        return new CrossCaseTypeSearchRequest(authorisedCaseTypeIds, originalSearchRequest.getSearchRequestJsonNode());
     }
 
-    private void filterCaseFieldsByAclAccess(CaseType caseType, CaseDetails caseDetails) {
-        JsonNode data = objectMapperService.convertObjectToJsonNode(caseDetails.getData());
-        JsonNode filteredData = accessControlService.filterCaseFieldsByAccess(data, caseType.getCaseFields(), getUserRoles(), CAN_READ);
-        caseDetails.setData(objectMapperService.convertJsonNodeToMap(filteredData));
+    private CaseSearchResult searchCasesAndFilterFieldsByAccess(List<CaseType> authorisedCaseTypes, CrossCaseTypeSearchRequest authorisedSearchRequest) {
+        return Optional.of(authorisedCaseTypes)
```

Comment:
I think this is a bit contrived. It takes some time to figure out what's going on. This would be simpler:
``` if (authorisedCaseTypes.isEmpty()) {
            return CaseSearchResult.EMPTY;
        }

        CaseSearchResult result = caseSearchOperation.execute(authorisedSearchRequest);
        filterFieldsByAccess(authorisedCaseTypes, result.getCases());
        return result;```

Reply:
👍